### PR TITLE
feat: migrate less to token for Checkbox

### PR DESCRIPTION
**Pull request by poyiding**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
ref [#41884](https://github.com/ant-design/ant-design/issues/41884)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     migrate less to token for Checkbox      |
| 🇨🇳 Chinese |     Checkbox 组件less变量迁移      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8d7fece</samp>

Moved `checkboxSize` from less variable to component token for `Checkbox` component. Updated the migration guide for less variables to component tokens in both English and Chinese documentation.

### <samp>🤖 Generated by Copilot at 8d7fece</samp>

* Move `checkboxSize` property from `CheckboxToken` to `ComponentToken` interface to enable user customization of checkbox size through `componentConfig` prop ([link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL6-R11))
* Remove `checkboxSize` property from `getStyle` function and use component token instead to calculate checkbox width and height ([link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL292), [link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL298-R304))
* Modify `genComponentStyleHook` function to accept a function that returns component token based on style token, and use it to derive `checkboxSize` from `controlInteractiveSize` global variable ([link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL298-R304))
* Add `Checkbox` component to migration guide for less variables to component tokens in English and Chinese documentation, and explain the deprecated or changed less variables and the new component token API (`docs/react/migrate-less-variables.en-US.md`, `docs/react/migrate-less-variables.zh-CN.md`, [link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aR39-R51), [link](https://github.com/ant-design/ant-design/pull/42097/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R51))

### CODE REVIEW

**Comment by MadCcc** [comment]
PR body 里 changelog 填一下

**Comment by MadCcc** [comment]
lint 有问题，类型需要改一下，可以参照其他的组件，checkbox 的 ComponentToken 这个类型没有使用过大概

**Comment by poyiding** [comment]
> lint 有问题，类型需要改一下，可以参照其他的组件，checkbox 的 ComponentToken 这个类型没有使用过大概 参考：https://github.com/ant-design/ant-design/pull/42083/files#diff-1d8c7d2026da8b02ccc59970cea90849086b133e4b8f55d59ae475ac814d9e1bR95


需要 @MadCcc 给genComponentStyleHooks 加上组件依赖

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -36,3 +36,16 @@ title: less 变量迁移 Component Token
...
+
+## Checkbox 多选框
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@checkbox-size` | `controlInteractiveSize` | - |
+| `@checkbox-color` | `colorPrimary` | - |
+| `@checkbox-check-color` | `colorWhite` | - |
+| `@checkbox-check-bg` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:48]: 这个可以加回来

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -36,3 +36,16 @@ title: less 变量迁移 Component Token
 | `@modal-confirm-body-padding` | - | 由于样式变化已废弃 |
 | `@modal-confirm-title-font-size` | `modalHeaderTitleFontSize` | - |
 | `@modal-border-radius` | `borderRadiusLG` | - |
+
+## Checkbox 多选框
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@checkbox-size` | `controlInteractiveSize` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:45]: 这个这样的话就可以不用把 checkboxSize 移到 ComponentToken 里去了

**Review by MadCcc — APPROVED** [review]

**Pull request merged**